src/components/report-viewer/report-viewer.tsx

```diff
@@ -8,104 +8,107 @@ import { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import Grid from '@mui/material/Grid';
 import LogTable from './log-table';
 import { mapReportsTree } from '../../utils/report/report-tree.mapper';
-import { useDispatch } from 'react-redux';
 import { VirtualizedTreeview } from './virtualized-treeview';
 import { ReportItem } from './treeview-item';
-import { Report, ReportLog, ReportTree, ReportType, SeverityLevel } from 'utils/report/report.type';
+import { Report, ReportLog, ReportTree, ReportType, SelectedReportLog, SeverityLevel } from 'utils/report/report.type';
+import { GLOBAL_REPORT_NODE_LABEL } from '../../utils/report/report.constant';
 
 type ReportViewerProps = { report: Report; reportType: string; severities: SeverityLevel[] | undefined };
 
-export default function ReportViewer({ report, reportType, severities }: ReportViewerProps) {
-    const dispatch = useDispatch();
+const DEFAULT_CONTAINER_HEIGHT_OFFSET = 170; // The value 170px is fine, but leaves a gap below the report.
 
+export default function ReportViewer({ report, reportType, severities = [] }: Readonly<ReportViewerProps>) {
     const [expandedTreeReports, setExpandedTreeReports] = useState<string[]>([]);
     const [highlightedReportId, setHighlightedReportId] = useState<string>();
-    const [reportVerticalPositionFromTop, setReportVerticalPositionFromTop] = useState<number | undefined>(undefined);
+    const [reportVerticalPositionFromTop, setReportVerticalPositionFromTop] = useState<number>(
+        DEFAULT_CONTAINER_HEIGHT_OFFSET
+    );
 
-    const [selectedReportId, setSelectedReportId] = useState(report?.id);
-    const [selectedReportType, setSelectedReportType] = useState<ReportType>();
+    const [selectedReport, setSelectedReport] = useState<SelectedReportLog>({
+        id: report.id,
+        type: report.message === GLOBAL_REPORT_NODE_LABEL ? ReportType.GLOBAL : ReportType.NODE,
+    });
 
     const reportTree = useMemo(() => mapReportsTree(report), [report]);
     const reportTreeMap = useRef<Record<string, ReportTree>>({});
 
-    const mapReportsById = useCallback((item: ReportTree) => {
-        reportTreeMap.current[item.id] = item;
-        for (let subReport of item.subReports) {
-            mapReportsById(subReport);
-        }
-    }, []);
+    const computeReportTreeMap = useCallback(() => {
```

Comment:
Maybe just deifne reportTreeMap as a useMemo ? Depending only on reportTree ?

Reply:
No problem, I did so initially but reverted to keep callbacks dependency arrays empty.
In the end it recomputes only when the whole tree changes so it's no big deal